Update release notes for k8s 1.16 version

evrardjp opened the pull request:
For release 4.1.0

Commit:
Import release notes for k8s 1.16 version

nkoranova committed:
Update attributes.

nkoranova committed:
Update Kubernetes changes info and instructions on what to do with deprecated apis.

nkoranova committed:
Specify kured revsion in attributes.

Commit:
Update Attributes of 4.1.0 release to update cri-o version

Commit:
Rephrase to avoid ambiguities

Without this, the content might become outdated on the new
versions. This makes the text definitive.

Commit:
Improve readability of the release notes

Without this patch, one user could read the release note, and
completely not know that manual intervention need to happen,
nor its high level impact.

This clarifies it by refering to the upgrade notes from the
release note of the version. The upgrades notes are then sorted
in the chronological order of things that need to happen.

Commit:
Make 4.1.0 upgrade details at the right location

Without this patch, all the details about 4.1.0 are in 4.0.3
section.

This should fix it.

Commit:
Fixup of the patch 959f988366ef7cfa7d31bc756983a24c10531648

In that patch, documentation of the 1.16 has removed 1.15 release
note details, and as incorrectly changed the title of the section.

This should fix it, by ensuring both details are incorporated.

Commit:
First instructions for kubernetes cluster upgrade

davidcassany commented on adoc/MAIN.release-notes.adoc:
```diff
@@ -15,7 +15,71 @@ The most recent version of the Release Notes is available online at https://www.
+. Check upgrades at the node level
+  ``skuba node upgrade plan <nodename>``
+. Upgrade all control plane node, one by one, by running
+  ``skuba node upgrade apply`` on each of them.
```
Isn't it mandatory to upgrade master nodes first? I think so

evrardjp replied:
Yes. That's what was written as "control plane". To be consistent with other places I moved to a "master"/"worker" terminology in the last patchset.

davidcassany reviewed (commented):
At glance I just have a comment about nodes upgrade order

Commit:
Replace control plane term with master node

Without this patch, it could be unclear what "control plane"
node means. "kubelet" could technically be considered as part
of the control plane if we think a little outside the box.

Let's make this clear. We are talking about "nodes" here, and the
"master" nodes come before the "worker" nodes.

Commit:
Remove duplicate information

nkoranova committed:
Fix ordered list.

nkoranova committed:
Fix internal cross references.

nkoranova committed:
Unify heading capitalization.

evrardjp commented:
Could you remove the label On-Hold please? It should be okay to review now.

Commit:
Add link to admin guide for updating

evrardjp commented:
Thanks for the update. Looks better indeed! :)

Itxaka reviewed (commented):
lgtm +1

ereslibre reviewed (approved):
LGTM, thanks @evrardjp
